server/src/main/java/org/opensearch/action/search/QueryInsightExporter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.common.unit.TimeValue;
+
+import java.util.List;
+
+/**
+ * Simple abstract class to export data collected by search query analyzers
+ * <p>
+ * Mainly for use within the Query Insight framework
+ *
+ * @opensearch.internal
+ */
+abstract class QueryInsightExporter<T extends SearchQueryRecord<?>> {
+
+    private boolean enabled = false;
+
+    /** The export interval of this exporter, default to 60 seconds */
+    private TimeValue exportInterval = TimeValue.timeValueSeconds(60);
+
+    /**
+     * Initial set up the exporter
+     */
+    abstract void setup() throws Exception;
```

Comment:
why do we need setup()? can't it be part of constructor of individual implementations?